app/workers/generate_test_applications.rb

```diff
@@ -152,6 +159,30 @@ def courses_from_cycle(year)
     courses.distinct
   end
 
+  def undergraduate_courses_from_cycle(year)
+    recruitment_cycle_year = [RecruitmentCycle.current_year, year].min
+    courses = Course
+      .teacher_degree_apprenticeship
+      .with_course_options
+      .in_cycle(recruitment_cycle_year)
+
+    if courses.count.zero?
+      provider = Provider.all.sample(10).sample || FactoryBot.create(:provider)
```

Comment:
I don't care what provider is created those applications and at this point I suspect each developer has your own provider favorite that test locally on QA. Let me know if you have strong opinions about this.

Reply:
I tend to use provider code 1TZ becuase they have primary and secondary courses by default. Only saying as you asked.

Reply:
@inulty-dfe I pushed one commit using the 1TZ as first try and if there isn't one, they will go to the next or statement and select at random.